src/vm/Components/FileHandlers/Reader.java

```diff
@@ -0,0 +1,50 @@
+package vm.Components.FileHandlers;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Scanner;
+
+public class Reader {
+
+    private FileReader fileReader;
+    private BufferedReader bufferedReader;
+    private String filename;
+
+    public Reader(String filename) throws FileNotFoundException {
+        this.filename = filename;
+        Path path = Path.of("src", "resources", filename);
+        fileReader = new FileReader(path.toAbsolutePath().toString());
+        bufferedReader = new BufferedReader(fileReader);
+    }
+
+    public String readLine() {
+        String data = "";
+        try {
+            data = bufferedReader.readLine();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return data;
+    }
+
+    public String readFile() throws FileNotFoundException {
```

Comment:
esse método ta sendo utilizado? não achei referência para ele 

Mas se estiver, acho que seria legal rever essa ideia de declarar o path duas vezes, aqui e no construtor. 
Gostei da ideia de já no construtor lançar a excessão de file not found, pois previne de ser instanciado um Reader que não será válido quando o arquivo passado não existir, então acho que poderia manter lá. Boa sacada!

Reply:
Não ta sendo utilizado ainda. Isso faz parte do que eu ia propor na segunda-feira sobre centralizar a manipulação de arquivos em um package só (e leitura e escrita também em uma classe só). Esse método é o teu la do sourceCodeHandler que eu passei pra aí e, caso todos estivessem de acordo, eu ia passar a instanciar ele ao invés daquele outro método pra ler arquivos. O detalhe é que eu não deveria ter commitado ele ainda kkkkkkkkkk

Reply:
sobre o path foi mal minha mesmo. No processo de passar teu código pra lá eu redefini por descuido meu. Vou subir um fix em breve se não for o caso de botar em pratica a minha ideia inicial de centralizar operações de IO ali

Reply:
ahh beleza

Reply:
fixed on commit https://github.com/guivahl/programacao-de-sistemas/commit/102987a69b97998bc4232bc1f3d19efae4e7d2c8